### PR TITLE
Cfg cluster op group

### PR DESCRIPTION
**Pull request by gabemontero**
per recent aos-devel announce from clayton ... moving to config.openshift.io

/assign @bparees 

@openshift/sig-developer-experience fyi / ptal

### CODE REVIEW

**Comment by gabemontero** [comment]
another test of the new e2e job @bparees :-)

**Comment by gabemontero** [comment]
oops forgot the unit test

**Comment by bparees** [comment]
@gabemontero since we have that awesome new e2e job, please add a test that confirms the clusteroperator object is getting created w/ appropriate status information.

**Comment by gabemontero** [comment]
assuming I get a clean run as-is @bparees, sure

**Comment by gabemontero** [comment]
aws nat pain on e2e-aws:

```
aws_eip.nat_eip.2: Error creating EIP: AddressLimitExceeded: The maximum number of addresses has been reached.
	status code: 400, request id: b28de6e6-5933-478b-a950-dad7d8006a2f
* module.vpc.aws_eip.nat_eip[3]: 1 error(s) occurred:
```

**Comment by gabemontero** [comment]
same nat pain with our new e2e job ... posted on forum-testplatform

**Comment by gabemontero** [comment]
the new e2e test failed on a timeout ... it never found the ruby imagestream

i have the new test ... hopefully I can debug what is up with the existing test while I vet out the new one

**Comment by gabemontero** [comment]
ah ha:
```
time="2018-11-07T19:05:44Z" level=error msg="error updating cluster operator status: failed to get cluster operator resource openshift-cluster-samples-operator/openshift-samples: clusterversions.config.openshift.io \"openshift-samples\" is forbidden: User \"system:serviceaccount:openshift-cluster-samples-operator:cluster-samples-operator\" cannot get clusterversions.config.openshift.io at the cluster scope: no RBAC policy matched"
```
more changes needed ....

**Comment by gabemontero** [comment]
A bit stuck ...my email to aos-devel:

Ok, after

- updating my operator's clusterrole rbac's to deal with `ClusterVersion`

- adding `k8sutil.AddToSDKScheme(ocfgapi.AddToScheme)` to register the new type with the operator sdk (where `ocfgapi` is ` "github.com/openshift/cluster-version-operator/pkg/apis/config.openshift.io/v1"`) after getting unknown type messages from the operator sdk

I can't seem to get past this error when I attempt to get/update/create my `ClusterVersion`

```
"error updating cluster operator status: failed to unmarshal the retrieved data: failed to decode json data with gvk(config.openshift.io/v1, Kind=ClusterVersion): v1.ClusterVersion.Spec: v1.ClusterVersionSpec.Upstream: DesiredUpdate: ClusterID: unmarshalerDecoder: invalid UUID length: 0, error found in #10 byte of ...|sterID\":\"\",\"desiredU|..., bigger context ...|723e82c3772e\"},\"spec\":{\"channel\":\"\",\"clusterID\":\"\",\"desiredUpdate\":null,\"upstream\":null}}\n|..."
```
The comment for the DesiredUpdate field notes it is optional.

Any insights on how to make the operator sdk work with `ClusterVersion`?  Or should I be using a different client?

I do get the following ClusterVersion (with the status missing as a result of the error I previously noted)
```
apiVersion: config.openshift.io/v1
kind: ClusterVersion
metadata:
  creationTimestamp: 2018-11-07T19:40:06Z
  generation: 1
  name: openshift-samples
  resourceVersion: "218197"
  selfLink: /apis/config.openshift.io/v1/clusterversions/openshift-samples
  uid: edb58aeb-e2c4-11e8-b436-723e82c3772e
spec:
  channel: ""
  clusterID: ""
  desiredUpdate: null
  upstream: null
```

**Comment by bparees** [comment]
@gabemontero have you bumped openshift/api and openshift/client-go in this repo?

**Comment by bparees** [comment]
(actually that may not matter)

File: deploy/clusteroperator-crd.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
-  name: clusteroperators.operatorstatus.openshift.io
+  name: clusterversions.operatorstatus.openshift.io
```
Comment (bparees) [inline deploy/clusteroperator-crd.yaml]: clusterversions.config.openshift.io ?

Comment (bparees) [reply deploy/clusteroperator-crd.yaml]: (honestly i dunno that we need to keep this file around anymore, it was a useful development tool when we stood up clusters manually but now that the installer is the only way, we can rely on the installer to have created this CRD in all our clusters)

File: deploy/03-rbac.yaml [hunk 2]
```diff
@@ -40,9 +40,9 @@ rules:
   verbs:
   - "*"
 - apiGroups:
-  - "operatorstatus.openshift.io"
+  - "config.openshift.io"
   resources:
-  - clusteroperators
+  - clusterversions
```
Comment (bparees) [inline deploy/03-rbac.yaml]: i don't think this was supposed to change.  the resource type is still clusteroperators.

File: deploy/clusteroperator-crd.yaml (hunk 1 above)
Comment (gabemontero) [reply deploy/clusteroperator-crd.yaml]: yeah I just as soon delete it ... will do that when I circle back to this PR (pending response to my outstanding client question I've circled back to implementing the progressing condition for image streams)

**Comment by gabemontero** [comment]
total mix up on my part ... @bparees got me back on track ... waiting on @dmage 's https://github.com/openshift/cluster-version-operator/pull/48

**Pull request closed without merging**